research-software-directory/RSD-as-a-service
# 418 software keywords filter

# Add keywords filter to software page

Closes #418 

Changes proposed in this pull request:
* User can filter on keywords on the software overview page. As the keywords are "free" values user needs fist to search for keywords he wants to add to filter
* Filtering on keywords is using AND "binding" between selected keywords.
* The filters are applied to url and can be saved or used to let other user land on the filtered software page.

How to test:

* `docker-compose down --volumes`: to remove old data structure
* `docker-compose build` to build new solution
* `docker-compose up` to start
* optional: run data migration
* on the software overview page you should see filter icon next to search input
![image](https://user-images.githubusercontent.com/9204081/181043539-0b8d1123-d15e-439d-a320-64a263092714.png)
* to add keyword you first type at least one letter of the keyword, a for example. The list of available keywords will be shown and you can select one to add. After you added few keywords click on apply button and the filter should be applied
* If you hover over the filter after it is active it will show you keywords used
![image](https://user-images.githubusercontent.com/9204081/181044787-84f973bf-328a-4745-872f-3df7e3f687da.png)
* The filter is implemented in the url too, so you can use the url to save filter selection. Copy the url, open the new tab and paste the url in the adressbar. It should open new page with filters applied.
* The CLEAR button in the filter drop-down will remove the filter.

PR Checklist:
